Quan_Ly_San_Pham_Luu_Ra_File_Nhi_Phan/src/com/codegym/DatabaseProduction.java

```diff
@@ -0,0 +1,124 @@
+                }
+                out.format(
+                        "----------------------------------------------------------------------------------------------------------------------");
+            } finally {
+                out.close();
+                in.close();
+            }
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        } catch (ClassNotFoundException ex) {
+            ex.printStackTrace();
+        }
+    }
+
+    public boolean findProduct(String name) {
+        String resultPath = "/root/Documents/Codegym_Projects/Module_2/Quan_Ly_San_Pham_Luu_Ra_File_Nhi_Phan/src/com/codegym/ProductionExtractTable.txt";
+        ByteArrayOutputStream bount = new ByteArrayOutputStream();
+        try {
+            ObjectOutputStream out = new ObjectOutputStream(bount);
+            ObjectInputStream in = null;
+            PrintWriter out2 = new PrintWriter(new File(resultPath).getAbsoluteFile());
+            try {
+                for (T item : this.storageProduction) {
+                    if (item.getName().equals(name)) {
+                        out.writeObject(item);
+                        in = new ObjectInputStream(new ByteArrayInputStream(bount.toByteArray()));
+                        
+                        Production production = (Production) in.readObject();
+
+                        out2.format("%70s\n", "KẾT QUẢ TÌM KIẾM");
+                        out2.format("%5s %20s %20s %20s %10s %15s", "Id", "Tên Sản Phẩm", "Mã Sản Phẩm",
+                                "Hãng Sản Xuất", "Giá", "Số Lượng\n");
+                        out2.format(
+                                "----------------------------------------------------------------------------------------------------------------------\n");
+                        out2.format("%5s", production.getId());
+                        out2.format("%20s", production.getName());
+                        out2.format("%30s", production.getCode());
+                        out2.format("%25s", production.getOriginal());
+                        out2.format("%17s", production.getPrice());
+                        out2.format("%15s\n", production.getQuantity());
+                        out2.format(
+                                "----------------------------------------------------------------------------------------------------------------------\n");
+
+                        return true;
+                    }
+                }
+                out2.format("%20s", "Sản phẩm bạn tìm kiếm không có trong danh sách sản phẩm !!!");
+                return false;
+
+            } finally {
+                out.close();
+                if (in != null){
+                    in.close();
+                }
+                out2.close();
+            }
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        } catch (ClassNotFoundException ex) {
+            ex.printStackTrace();
+        }
+
+        return false;
```

Comment:
hàm findProduct kiểu boolean thì chỉ nên kiểm tra dữ liệu và trả về true false, không nên để các câu lệnh in ra màn hình ở đây

Reply:
Ok em, anh để trả về kiểu đúng sai mà lại đưa thêm hiển thị sản phẩm đúng là không hợp lý.